--- xs/Sort_wrap.c	2010-05-25 14:46:37.000000000 +0200
+++ xs/Sort_wrap.c.patched	2010-05-26 13:22:06.000000000 +0200
@@ -1915,10 +1915,10 @@
       
       tempav = (AV*)SvRV(ST(0));
       len = av_len(tempav);
-      arg1 = (size_t *) malloc((len+1)*sizeof(size_t));
+      arg1 = (size_t *) malloc((len+2)*sizeof(size_t));
       for (i = 0; i <= len; i++) {
         tv = av_fetch(tempav, i, 0);
-        arg1[i] = SvIV(*tv);
+        arg1[i] = (size_t) SvIV(*tv);
       }
     }
     {
